src/main/java/CLIParser.java

```diff
@@ -0,0 +1,42 @@
+import java.util.Arrays;
+
+public class CLIParser {
+  private static final String DIRECTORY_FLAG = "-dir";
+  private static final String PORT_FLAG = "-port";
+
+  private String[] args;
+  private int port;
+  private String directory;
+
+  public CLIParser(String[] args) {
+    this.args = args;
+    parseArguments();
```

Comment:
Let's move the actual parsing of the arguments out of the constructor and into `getPortNumber` and `getDirectory`. Ideally, constructors shouldn't contain any logic and we should only be setting properties with given arguments.

Imagine if you were to package this library for public use and somehow `parseArguments` throws an exception. It'd be very confusing for a user to get an exception on object creation.

Reply:
Got it. In order to avoid having two nearly identical for loops, I've refactored the looping out to `getArgumentByFlag(String flag)`.